ext/cl_khr_command_buffer.asciidoc

```diff
@@ -1516,6 +1523,172 @@ New errors:
   support {CL_COMMAND_BUFFER_CAPABILITY_DEVICE_SIDE_ENQUEUE_KHR} and _kernel_
   contains a kernel-enqueue call.
 
+
+The function
+
+include::{generated}/api/protos/clCommandSVMMemcpyKHR.txt[]
+
+Records a command to do an SVM memcpy operation.
+
+_command_buffer_ Refers to a valid command-buffer object.
+
+_command_queue_ Specifies the command-queue the command will be recorded to.
+Parameter is unused by this extension as only a single command-queue is
+supported and **must** be `NULL`.
+
+_dst_ptr_ is the pointer to a host or SVM memory allocation where data is
+copied to.
+
+_src_ptr_ is the pointer to a host or SVM memory allocation where data is
+copied from.
```

Comment:
I think we should think carefully whether we want to support memcpys to or from a host allocation, at least when the device executing the command buffer does not support system SVM.  Supporting arbitrary host allocations could be expensive for some devices and IIRC this was one of the reasons why we do not allow equivalents of `clEnqueueReadBuffer` or `clEnqueueWriteImage` to be recorded in a command buffer.

Can we restrict this to SVM memory allocations only, unless the device supports system SVM?

Reply:
That's a good point, supporting host memory was problematic for one of our customers, so we didn't want equivalents. Omitting the entry-points also reduced the scope of the extension so that we could release something publicly earlier.

If I recall, rather than omitting the use of host memory in command-buffers @pjaaskel asked for it to be optional. Maybe that's something we want to revisit before the extension moves from provisional to final. Extension could have a device query to report if you can use host memory in SVM memcpy/memfill and also record `clEnqueueReadBuffer` type commands.

Reply:
Good points. Yes, the capabilities of SVM commands that can be buffered should naturally match to the level of SVM supported.

I planned to submit host interaction and the read/write buffer commands as separate layered extensions on top, but naturally making them optional parts of the extension would be simpler. We try to fulfil the needs of CUDA/HIP graph with these enhancements. It's ongoing work.